### PR TITLE
Routing queue config & default subscriber

### PR DESCRIPTION
**Pull request by xcchang**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

**Review by shreayp — APPROVED** [review]
LGTM

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java [hunk 1]
```diff
@@ -131,6 +133,13 @@ public static ReplicationSubscriber getDefaultLogicalGroupSubscriber() {
                 .build();
     }
 
+    public static ReplicationSubscriber getDefaultRoutingQueueSubscriber() {
+        return ReplicationSubscriber.newBuilder()
+                .setClientName(DEFAULT_ROUTING_QUEUE_CLIENT)
+                .setModel(ReplicationModel.LOGICAL_GROUPS)
```
Comment (pankti-m) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java:139]: model is not Logical Groups

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java [hunk 2]
```diff
@@ -458,14 +483,18 @@ public CorfuStoreMetadata.Timestamp onClientListenerResume() {
...
-        switch(subscriber.getModel()) {
+    public UUID getOpaqueStreamToTrack(LogReplicationSession session) {
+        switch(session.getSubscriber().getModel()) {
             case FULL_TABLE:
                 return ObjectsView.getLogReplicatorStreamId();
             case LOGICAL_GROUPS:
-                return ObjectsView.getLogicalGroupStreamTagInfo(subscriber.getClientName()).getStreamId();
+                return ObjectsView.getLogicalGroupStreamTagInfo(session.getSubscriber().getClientName()).getStreamId();
+            case ROUTING_QUEUES:
+                String logEntrySyncStreamTag = ((LogReplicationRoutingQueueConfig) sessionToConfigMap.get(session))
```
Comment (pankti-m) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java:493]: there will be a separate opaque stream to track if in snapshot sync.  We will now use the same approach to find entries to replicate in snapshot sync also.  It cannot be assumed that this method will get invoked in LogEntry Sync only

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java [hunk 3]
```diff
@@ -199,11 +209,26 @@ public void generateConfig(Set<LogReplicationSession> sessions) {
...
     }
 
+    private void generateRoutingQueueConfig(LogReplicationSession session) {
+        if (!sessionToConfigMap.containsKey(session)) {
+            LogReplicationRoutingQueueConfig routingQueueConfig =
+                    new LogReplicationRoutingQueueConfig(session, serverContext);
+            sessionToConfigMap.put(session, routingQueueConfig);
+            log.warn("Routing queue session {} config generated.", session);
+        } else {
+            log.warn("Routing queue config for session {} already exists!", session);
```
Comment (souravgarg) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java:228]: do you need warn log in both if-else case?

**Review by shama358 — APPROVED** [review]

**Review by pankti-m — APPROVED** [review]

**Pull request merged**